Sources/SourceKitLSP/SourceKitLSPServer.swift

```diff
@@ -931,8 +947,8 @@ extension SourceKitLSPServer {
       options: options,
       indexOptions: self.options.indexOptions,
       indexTaskScheduler: indexTaskScheduler,
-      indexProcessDidProduceResult: { [weak self] in
-        self?.indexTaskDidProduceResult($0)
+      logMessageToIndexLog: { taskID, message in
+        self.logMessageToIndexLog(taskID: taskID, message: message)
```

Comment:
No `weak self`?

Reply:
Also, should have made this comment sooner, but should we consider using a delegate or something to cut down on the amount of closures we have to pass around?